content/MANUAL.md

```diff
@@ -1,61 +1,316 @@
+GitHub Pages is a free hosting service that takes HTML, CSS, and JavaScript files straight from a repository on GitHub, optionally runs the files through a build process, and publishes a website.
+
+#### Setting Up GitHub Pages for Your Repository
+
+1. **Enable GitHub Pages in your repository**:
+   - Go to your repository on GitHub
+   - Navigate to Settings > Pages
+   - Set the Source to "GitHub Actions" (recommended) or "Deploy from a branch"
+   
+2. **Configure deployment in your project**:
+   - In `Directory.UserConfig.props`, ensure `<GitHubDeployment>true</GitHubDeployment>` is set
+   - Update the `<PackageProjectUrl>` to match your GitHub repository URL
+
+#### Deploying Your Demo
+
+The demo can be automatically published to GitHub Pages:
+
+```bash
+./build.sh gh-pages    # Unix
+build.cmd gh-pages     # Windows
 ```
+
+This command:
+1. Builds your binding project
+2. Builds the demo application with production settings
+3. Deploys the built files to the `gh-pages` branch or via GitHub Actions
+4. Makes your demo available at `https://your-username.github.io/your-repository`
+
+#### Tips for Effective GitHub Pages Demos
+
+1. **Update the demo's index.html title and metadata** to properly describe your binding
+   ```html
+   <!-- demo/index.html -->
+   <title>My Awesome Binding - Interactive Demo</title>
+   <meta name="description" content="Interactive demo for MyBinding - a Fable binding for XYZ library">
+   ```
+
+2. **Add usage examples** to make your demo showcase real-world use cases
+
+3. **Include documentation links** in your demo to help users understand how to use your binding
+
+4. **Test your demo thoroughly** before deploying:
+   ```bash
+   # Test production build locally before deploying
+   cd demo
+   npm run build
+   npm run preview
+   ```
+
+5. **Customize the base URL** if needed in `vite.config.ts`:
+   ```typescript
+   export default defineConfig({
+     base: '/your-repository-name/',
+     // other config...
+   })
+   ```
+
+#### Troubleshooting GitHub Pages
+
+- **Deployment fails**: Check your GitHub repository settings and ensure GitHub Pages is properly configured
+- **CSS/JS not loading**: Verify the `base` path in vite.config.ts matches your repository name
+- **404 errors**: Make sure all asset paths are relative or respect the base URL
+- **Custom domain**: If using a custom domain, add it in your GitHub repository settings and create a CNAME file in the demo directory
+
+## Troubleshooting
+
+### Common Issues
+
+1. **Build script not executable** (Unix)
+   ```bash
+   chmod +x build.sh
+   ```
+
+2. **Missing tools**
+   ```bash
+   dotnet tool restore
+   ```
+
+3. **JavaScript dependencies issues**
+   ```bash
+   cd demo
+   rm -rf node_modules
+   npm install
+   ```
+
+4. **Release validation fails**
+   - Check that all required properties are set in Directory.UserConfig.props
+   - Ensure your git workspace is clean
+   - Verify that your commit messages follow the convention
```

Comment:
All good feedback, but errors should be reported by the `build` process with explanation about the issues so I am not sure we need a dedicated `Troubleshooting` for that.

For example, you can't make a release if properties have not been set:

https://github.com/glutinum-org/Glutinum.Template/blob/3600628ca7f4478f34560f08683f6e9a03c76e66/content/Directory.Build.props#L32-L37